### PR TITLE
doc: allow nrfxlib to link to nrf with intersphinx

### PR DESCRIPTION
**Pull request by utzig**
Add nrf objects.inv to nrfxlib intersphinx mapping to allow automatic link resolution.

JIRA: NCSDK-3934

### CODE REVIEW

**Comment by utzig** [comment]
This is complimentary to https://github.com/nrfconnect/sdk-nrfxlib/pull/218. While this works it requires a two-step build of nrfxlib, first building `nrfxlib`, then `nrf`, then cleaning `nrfxlib` and building `nrfxlib` again, which is awful. A decent process would be to build the `objects.inv` files without generating the final html output for both projects, followed by the final html for both.

**Comment by utzig** [comment]
Update from commit messages:

Add a new Sphinx extension to be used as a builder, which when run generates only an inventory file (objects.inv). This can be used to break a build into two steps when there are inter-references between projects:

1) First run a build step in the first projects, to generate an objects.inv using the inventory builder. The objects.inv can then be used as input for a second project.
2) Build the second project with the html builder, which will generate the html output along with the objects.inv file, and will fix any intersphinx external references using the inventory file from the first step.
3) Build the initial project again, this time using the html builder, using the objects.inv file from the second project.

* Add a new build step, nrfxlib-inventory, which will use the inventory builder Sphinx extension.
* Optionall add the nrf objects.inv to nrfxlib intersphinx mapping to allow for automatic link resolution.

To build nrf + nrfxlib now must be run in three steps:

1) ninja nrfxlib-inventory
2) ninja nrf
3) ninja nrfxlib

**Comment by utzig** [comment]
I suppose this will also require updating the build process docs, CI, etc!

**Comment by carlescufi** [comment]
> I suppose this will also require updating the build process docs, CI, etc!

Yes, could you start by updating the instructions in the .rst file itself in this repo? This can be done in this PR

**Comment by utzig** [comment]
> Yes, could you start by updating the instructions in the .rst file itself in this repo? This can be done in this PR

Sure can!

**Comment by utzig** [comment]
> This PR requires a change to the CI before it can be merged, and it will make other PRs fail until they are rebased.

I am submitting the CI changes as well, and would rather merge them before this PR. The way I wrote it, the build script checks if the nrfxlib inventory needs to be built, and acts accordingly, so it should work both with PRs that don't need the inventory and this one (and newer ones afterwards).

File: doc/nrf/doc_build.rst [hunk 1]
```diff
@@ -86,24 +86,31 @@ Complete the following steps to build the documentation output:
...
 
+#. Run ninja to build the nrfxlib documentation:
+
+        .. code-block:: console
+
+           ninja nrfxlib
+
 The documentation output is written to ``_build\html``. Double-click the ``index.html`` file to display the documentation in your browser.
 
 .. tip::
```
[the remark this answers is not in the document]
Comment (utzig) [reply doc/nrf/doc_build.rst:116]: Done!

File: doc/nrf/doc_build.rst [hunk 2]
```diff
@@ -86,30 +86,42 @@ Complete the following steps to build the documentation output:
...
-   If you modify or add RST files, you only need to rerun the steps that build the respective documentation: step 4 (if you did changes to Kconfig), step 5 (if you modified the Zephyr documentation), step 6 (if you modified the nrfxlib documentation), step 7 (if you modified the MCUboot documentation), or step 8 (if you modified the |NCS| documentation).
 
+   If you modify or add RST files, you don't need to rerun all steps. For simple
```
Comment (ru-fu) [inline doc/nrf/doc_build.rst:118]: A bit more nitpicking, sorry: Please format with one sentence per line (according to our style guide), and don't use "don't". ;)

Comment (utzig) [reply doc/nrf/doc_build.rst]: Done!

**Comment by utzig** [comment]
> I am submitting the CI changes as well, and would rather merge them before this PR. The way I wrote it, the build script checks if the nrfxlib inventory needs to be built, and acts accordingly, so it should work both with PRs that don't need the inventory and this one (and newer ones afterwards).

CI PR was merged, so this one should pass on the next build....

**Comment by carlescufi** [comment]
Please update the nrfxlib SHA in `west.yml` to 92075b5e10bbf1600dbb36d2a983194bcd4319a4

**Pull request merged**